contrib/epee/src/string_tools.cpp

```diff
@@ -141,52 +144,55 @@ namespace string_tools
 #ifdef _WIN32
   std::string get_current_module_path()
   {
-    char pname [5000] = {0};
-    GetModuleFileNameA( NULL, pname, sizeof(pname));
-    pname[sizeof(pname)-1] = 0; //be happy ;)
-    return pname;
+    std::wstring buf;
+    buf.resize(MAX_PATH);
+    do
+    {
+        unsigned int len = GetModuleFileNameW(NULL, &buf[0], static_cast< unsigned int >(buf.size()));
+        if (len == 0) {
+          DWORD error = GetLastError();
+          // Log and return empty string since don't do error control in this function
+          MFATAL("Error : " << std::system_category().message(error));
+          return {};
+        }
+        else if (len < buf.size())
+        {
+            buf.resize(len);
+            break;
+        }
+        buf.resize(buf.size() * 2);
+    }
+    while (buf.size() < 65536);
+    return boost::filesystem::path(buf).string();
   }
 #endif
-	
-    bool set_module_name_and_folder(const std::string& path_to_process_)
-	{
-    std::string path_to_process = path_to_process_;
+
+  void set_module_name_and_folder(const std::string& path_to_process_)
+  {
+    boost::filesystem::path path_to_process = path_to_process_;
+
 #ifdef _WIN32
     path_to_process = get_current_module_path();
 #endif 
-		std::string::size_type a = path_to_process.rfind( '\\' );
-		if(a == std::string::npos )
-		{
-			a = path_to_process.rfind( '/' );
-		}
-		if ( a != std::string::npos )
-		{	
-			get_current_module_name() = path_to_process.substr(a+1, path_to_process.size());
-			get_current_module_folder() = path_to_process.substr(0, a);
-			return true;
-		}else
-			return false;
 
-	}
+    get_current_module_name() = path_to_process.filename().string();
+    get_current_module_folder() = path_to_process.parent_path().string();
+  }
 
 	//----------------------------------------------------------------------------
-	bool trim_left(std::string& str)
-	{
-		for(std::string::iterator it = str.begin(); it!= str.end() && isspace(static_cast<unsigned char>(*it));)
-			str.erase(str.begin());
-			
-		return true;
-	}
-	//----------------------------------------------------------------------------
-	bool trim_right(std::string& str)
-	{
+  void trim_left(std::string& str)
+  {
+    boost::trim_left(str);
+    return;
```

Comment:
Should this redundant `return;` statement be removed? I believe the `void` function will return here anyway.

(If so, this is also the case in the `trim_right` function a few lines below.)